### ISSUE TITLE
think about the request-holder-filter for spring boot applications again

### ISSUE THREAD

**Issue by bennetelli** [togglz/togglz#760]
At the moment I am not sure if I would like that users needs to add the request-holder-filter entry in the application.properties. 
Maybe we should discuss this once again and flip the status, so only really edge case users like people who are building an application as a war file and deploying it to an external tomcat instead of using the internal one need to set that entry
and all the others (which I am expecting are far more) who are using Spring Boot as far jar containing everything don't have to set anything like this and the application is just running after they added the starter dependency.

that's the corresponding line to search for in the code: @ConditionalOnProperty(prefix = "togglz.web", name = "request-holder-filter", havingValue = "true")

**Comment by hennr** [comment]
Yep, I like the idea and would vote to have it enabled per Default.
It's quite a bummer when you are trying to use the header activation strategy for example and it simply doesn't work out of the box.

**Comment by PH1P5** [comment]
Hi all,

Is there a need for conditional initialization of the _HttpServletRequestHolderFilter_ within the spring boot starter?
If so I see two options:
- conditional on bean _ActivationStrategyProvider_, since the request holder is only used with the activation strategies
- conditional on web app, since this is a web application feature

wdyt?

**Comment by bennetelli** [comment]
> Yep, I like the idea and would vote to have it enabled per Default. It's quite a bummer when you are trying to use the header activation strategy for example and it simply doesn't work out of the box.

indeed. We should have realized this much earlier.

**Comment by hennr** [comment]
It seems that the default for the request-holder-filter was set to false via PR #537 which was merged to fix the issue #535
This renders the Header activation strategy useless per default which is a bad experience for users and also was not the goal discussed in #535 

I'd suggest to simply turn around the default for the property `togglz.web.request-holder-filter` to true like so:
`@ConditionalOnProperty(prefix = "togglz.web", name = "request-holder-filter", havingValue = "true", matchIfMissing = true)`

This way users would be able to use the Header AS per default and users with WAR deployments would still be able to disable the feature.
Opinions on this @bennetelli  @PH1P5 ?

**Comment by PH1P5** [comment]
I agree having request-holder-filter enabled by default.
